src/runtime/StaticStrings.cpp

```diff
@@ -26,92 +26,100 @@ namespace Escargot {
+    INIT_STATIC_STRING(stringTypeof, "typeof");
+    INIT_STATIC_STRING(stringSwitch, "switch");
+    INIT_STATIC_STRING(stringExport, "export");
+    INIT_STATIC_STRING(stringImport, "import");
+    INIT_STATIC_STRING(stringFinally, "finally");
+    INIT_STATIC_STRING(stringExtends, "extends");
+    INIT_STATIC_STRING(stringFunction, "function");
+    INIT_STATIC_STRING(stringContinue, "continue");
+    INIT_STATIC_STRING(stringDebugger, "debugger");
+    INIT_STATIC_STRING(stringInstanceof, "instanceof");
+    INIT_STATIC_STRING(stringImplements, "implements");
+    INIT_STATIC_STRING(stringInterface, "interface");
+    INIT_STATIC_STRING(stringPackage, "package");
+    INIT_STATIC_STRING(stringYield, "yield");
+    INIT_STATIC_STRING(stringLet, "let");
+    INIT_STATIC_STRING(stringError, "error");
+
+    INIT_STATIC_STRING(defaultRegExpString, "(?:)");
+
+    INIT_STATIC_STRING(get__proto__, "get __proto__");
+    INIT_STATIC_STRING(set__proto__, "set __proto__");
+    INIT_STATIC_STRING(getbyteLength, "get byteLength");
+    INIT_STATIC_STRING(getbyteOffset, "get byteOffset");
+    INIT_STATIC_STRING(getLength, "get length");
+    INIT_STATIC_STRING(getBuffer, "get buffer");
+    INIT_STATIC_STRING(getSize, "get size");
+    INIT_STATIC_STRING(getFlags, "get flags");
+    INIT_STATIC_STRING(getGlobal, "get global");
+    INIT_STATIC_STRING(getIgnoreCase, "get ignoreCase");
+    INIT_STATIC_STRING(getMultiline, "get multiline");
+    INIT_STATIC_STRING(getSource, "get source");
+    INIT_STATIC_STRING(getSticky, "get sticky");
+    INIT_STATIC_STRING(getUnicode, "get unicode");
+    INIT_STATIC_STRING(getSymbolSpecies, "get [Symbol.species]");
+    INIT_STATIC_STRING(getSymbolSearch, "get [Symbol.search]");
+    INIT_STATIC_STRING(getSymbolToStringTag, "get [Symbol.toStringTag]");
+    INIT_STATIC_STRING(symbolSplit, "[Symbol.split]");
+
+    INIT_STATIC_STRING($Ampersand, "$&");
+    INIT_STATIC_STRING($PlusSign, "$+");
+    INIT_STATIC_STRING($GraveAccent, "$`");
+    INIT_STATIC_STRING($Apostrophe, "$'");
+#undef INIT_STATIC_STRING
+
+#define INIT_STATIC_NUMBER(num) numbers[num].initStaticString(atomicStringMap, new ASCIIString(#num, strlen(#num)));
+    FOR_EACH_STATIC_NUMBER(INIT_STATIC_NUMBER)
+#undef INIT_STATIC_NUMBER
+
+    for (unsigned i = 0; i < ESCARGOT_ASCII_TABLE_MAX / 2; i++) {
+        const char c = (const char)i;
+        asciiTable[i].init(atomicStringMap, &c, 1);
+    }
+
+    for (unsigned i = ESCARGOT_ASCII_TABLE_MAX / 2; i < ESCARGOT_ASCII_TABLE_MAX; i++) {
         LChar buf[2];
         buf[0] = i;
         buf[1] = 0;
         asciiTable[i].init(atomicStringMap, buf, 1);
     }
 
-    for (unsigned i = 0; i < ESCARGOT_STRINGS_NUMBERS_MAX; i++) {
-        ASCIIStringData s = ::Escargot::dtoa(i);
-        numbers[i].init(atomicStringMap, s.data(), s.size());
-    }
-
 #define INIT_STATIC_STRING(name) name.init(atomicStringMap, #name, strlen(#name));
     FOR_EACH_STATIC_STRING(INIT_STATIC_STRING)
```

Comment:
@rerobika It seems that `AtomicString::initStaticString` could be used here too for initializing StaticStrings. How about your opinion?

Reply:
Sure, good catch!